### PR TITLE
Refactor rhobot cli & gocd 

### PR DESCRIPTION
**Pull request by frank0631**
moved the guts of rhobot into helper functions, move those helper functions out
separated high level, low level, and structures for gocd module.
latest version of gocd api.
added stubs for new healthcheck operations (less than, not equals...)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -32,13 +24,21 @@ before_install:
...
-    - go get github.com/Sirupsen/logrus
-    - go get github.com/urfave/cli
-    - go get -u github.com/davecgh/go-spew/spew
-    - go get github.com/lib/pq
-    - go get gopkg.in/yaml.v2
-    - go get github.com/flosch/pongo2
-    - go get gopkg.in/gomail.v2
+    - go get github.com/tools/godep
+    - godep restore
+    # - go get github.com/Sirupsen/logrus
```
Comment (tanderegg) [inline .travis.yml]: Can we just remove these commented out lines?

File: gocd/gocd.go [hunk 2]
```diff
@@ -2,350 +2,26 @@ package gocd
...
-
-	if resp.StatusCode != 200 {
-		err = fmt.Errorf("Bad response code: %d with response: %s", resp.StatusCode, body)
-		return
-	}
-
-	historyJSON, err = printPrettyJSON(body, "pipelineHistory")
-	return
-}
+// // History gets the run history of a pipeline of a given name exist. returns map
```
Comment (tanderegg) [inline gocd/gocd.go]: Same here, can we just remove this now?

File: gocd/gocd_util.go [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// Server a representstion of a GoCD server
```
Comment (tanderegg) [inline gocd/gocd_util.go]: Spelling/grammar :)

File: gocd/gocd_util.go [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+}
+
+// Partially generated by curl-to-Go: https://mholt.github.io/curl-to-go
+func (server Server) pipelineConfigPUT(pipeline Pipeline, etag string) (pipelineResult Pipeline, err error) {
+
+	pipelineName := pipeline.Name
+
+	payloadBytes, err := json.Marshal(pipeline)
+	if err != nil {
+		return
```
Comment (tanderegg) [inline gocd/gocd_util.go:87]: Should this error be logged here, or is that handled by the caller?

File: healthcheck/templateHealthcheck.go [hunk 5]
```diff
@@ -14,14 +14,15 @@ CREATE TABLE IF NOT EXISTS {{metadata.schema}}.{{metadata.table}}
   query text,
   executed text,
   expected text,
+	operation text,
```
Comment (tanderegg) [inline healthcheck/templateHealthcheck.go]: Extra indentation?

File: healthcheck/templateHealthcheck.go [hunk 6]
```diff
@@ -36,6 +37,7 @@ const TemplateHealthcheckHTML = `
             <th>Query</th>
             <th>Test Ran?</th>
             <th>Expected</th>
+						<th>Operation</th>
```
Comment (tanderegg) [inline healthcheck/templateHealthcheck.go]: Here too, too much indent?

**Review by tanderegg — CHANGES_REQUESTED** [review]
Some minor comments, otherwise looks good!

File: .travis.yml (hunk 1 above)
Comment (frank0631) [reply .travis.yml]: yup

File: gocd/gocd_util.go (hunk 3 above)
Comment (frank0631) [reply gocd/gocd_util.go]: golint has some rules dictating how exported types should be commented. They may not always be english friendly, but I'll keep it in mind when commenting future types/functions

File: gocd/gocd_util.go (hunk 4 above)
Comment (frank0631) [reply gocd/gocd_util.go:87]: this occurrence, as well as all the "curl-to-Go" generated functions, pass the errors up to the caller.

File: healthcheck/templateHealthcheck.go (hunk 5 above)
Comment (frank0631) [reply healthcheck/templateHealthcheck.go]: Invisible characters are the bane of my existence.

File: healthcheck/templateHealthcheck.go (hunk 6 above)
Comment (frank0631) [reply healthcheck/templateHealthcheck.go]: I don't know why my IDE (Atom) is not showing me these tabs in the right way

**Review by tanderegg — APPROVED** [review]

**Pull request merged**